snapshot.py

```diff
@@ -71,7 +72,10 @@ def read_stats(args):
 
     if args.ip == '':
         LOGGER.info('Loading local procfs files')
-        cmd = "sudo bash -c \"%s\"" % (cmd % (pids, pids))
+        if whoami().strip() != "root":
+            LOGGER.error("Requires root privileges to run locally")
+            sys.exit(1)
+        cmd = "bash -c \"%s\"" % (cmd % (pids, pids))
```

Comment:
Ok this is pretty reasonable. A "--non-root" option could be added in future to get just the subset of info available to the current user.